README.md

```diff
@@ -70,6 +70,30 @@ built client from the client repository. Please consult the client's
 documentation for instructions on building the client in a development
 environment.
 
+**Tip**: If you get a **permission denied** error when running `npm link`
```

Comment:
I think you make a lot of fair points.

I would say however that any explanations about the purpose and usage of common tools in a language's ecosystem should be brief - no more than a sentence or two, with a helpful link for people who haven't encountered them before. Even though we try to be careful about paring down the dependencies of the client and service, if we explained all the tools required to develop them in detail it would take up a lot of space and add a lot of noise for many readers.

If there is a particular feature of the tool which is less common that we recommend using (_npm link_ is a reasonable example), then a slightly more detailed explanation I think is OK.

> or even giving a potentially broken-in-future link to a page in the docs 

The other side is that any documentation we write, we also have to maintain. It could also get out of date when newer versions of the tool are released.